### DISCUSSION TITLE
Remaking my old Nim videos

### DISCUSSION THREAD

**Discussion by Kiloneie** [Kiloneie/kiloneie.github.io#3] [General]
I have been thinking about this idea for a while now. This holds especially true for the very first Nim video that i made of [Nim for Beginners #1 Introduction to Programming](https://youtu.be/5tVIsDYPClA). 

The first 30 or so videos have 2.5K resolution down scaled to 1080p, and i am pretty sure i did not use as good of a video quality settings that my more recent videos use. Also 30 fps(this matters only in extremely rare cases). I also fiddled with audio settings very often back then, and the results are in general MUCH worse than they are now. Some even had a noise gate incorrectly set, and with that, some sentences had their ending cut out without me realizing it. There was also WAY too loud keyboard noise, due to my mic being on the desk, further away from me than it was to the keyboard. Which is not glued to my headphones of about equal quality, but the distance is EXTREMELY important for picking up a lot less background noise of my mouse and keyboard, as well as the fact that i have returned to the usage of a noise gate, and FINALLY set it up correctly. It was used in the last 2x videos, the 2x that now use a lot of darkening to focus in on areas i am currently talking about. I set it up so nicely(spent like 2 hours on it), that i only had to cut out my breathing like 3x times TOTAL, and it didn't cause any unwanted effects.

The next video will also use the software version of the AV1 encoder, SVT-AV1 with custom settings that i spent about a day or more on... In order for the video quality to be as good as it is now, while decreasing the file size by about half. I need smaller file size for editing, too long of a length videos. By diving deeper into OBS, i have also upped the audio bitrate from 160 to 256 KB, and when trying it out, sometimes i could hear the difference on my HD 599 headphones(saying their name, because they are pretty good, and therefore matter in being able to hear the difference).

So with all the video quality improvements, audio improvements, in general how my videos are structured, the content, transitions, examples, having written versions, no more staring at empty VS Code, but reading along with me(99.99...% of all YT videos are scripted, there is NO point in hiding the fact)... My newest 2x videos, and my future ones which i will keep trying to improve, are just **so** much better than my old ones, that's it's a clear night and day difference going forth between them. And the very first video i made for Nim, is also outdated. I used the term GC/Garbage Collector which is often viewed as bad in the programming world, when i should of simply said memory management. And now we have ORC, which is really good with memory, and is a good selling point of Nim for embedded programming.

**TL DR:**
Anyways, ALL of the videos up to the first SDL video, require a remake. Especially the very first one. The remakes will try to keep a very similar video length and content, in order to improve as much as possible without spending a colossal amount of time on them. The examples shown in those videos will be mostly replaced with more unique and possibly better ones with even less ambiguity, and with any errors i have made fixed and or removed. They will obviously due to how i make my video scripts, also be getting written tutorial versions, which was the idea from the VERY START. That sadly didn't happen due to my inexperience at the time. I did make pretty good scripts in OneNote back then, but then i also transitioned to making them in comments in VS Code, because copying between the two would introduce a LOT of weird styling symbols and hidden tabs(a nightmare)... Then also, the video and audio quality would also drastically/dramatically improve. Old versions will remain on YT and will be set to `unlisted`, meaning anyone with the video link can view them. And that link will be posted in the new version's description, for nostalgia reasons, and MORE IMPORTANTLY, keep the view count up for the YT algorithm. They will also be removed from their playlists, and moved to a `Old video versions` playlist.

Now, if you got any additional ideas to my own, please feel free to share them :).
